### ISSUE TITLE
Find-Module not working inside function (as scheduled task)

### ISSUE THREAD

**Issue by jasonshave** [PowerShell/PowerShell#8675]
I've created a function to run "Find-Module" with some additional parameters. I can run the function interactively without issue. The results come back and are stored in a variable I can use. If I let the function run as a scheduled task there is no output. This is the only function out of thousands of lines of code and many other functions which run as scheduled tasks; all without issue.

Is there a known limitation or configuration necessary to get "Find-Module" to work inside a function which runs as a scheduled task?

**Comment by Northman-de** [comment]
`Find-Module psread* | Export-Clixml d:\temp\findmodule.xml`
Works for me. Do you use a proxy? Has the task account access to the proxy?

**Comment by anmenaga** [comment]
@jasonshave it would be interesting to see task details; how it is configured.

**Comment by jasonshave** [comment]
This has been resolved but remains an issue in principal. As it turns out, the registration of a `PSRepository` is a per/user action. My scheduled task runs under the `SYSTEM` context in Windows which means when the scheduled task runs, it can't find any user-created repository so the `Find-Module` command was working as expected.

I was testing an auto-updating feature of my PowerShell module and didn't want to post many updates to PS Gallery (as per the guidance about avoiding this behavior). Once I published my module for the first time to PSG, my auto-update function worked as desired.

[There is a feature request covering this matter.](https://github.com/PowerShell/PowerShellGet/issues/214)
